src/lib_context/recording/trace_stats_summary_pp.ml

```diff
@@ -343,6 +240,28 @@ module Table0 = struct
     | `Gc_custom_minor_max_size ->
         string_of_int s.header.gc_control.custom_minor_max_size
 
+  let extract_keys_from s set =
+    match s.header.config.message with
+    | None -> set
+    | Some json -> (
+        match parse_json json with
+        | Ok kv ->
+            List.fold_left
+              (fun acc (json_key, _) -> MsgSet.add json_key acc)
+              set
+              kv
+        | _ -> set)
+
+  let extract_all_keys summaries : [> `Key of string] list =
```

Comment:
Putting the type here highlight your wish, which is a good thing by itself. However the compiler will only use it to extend to lower bound. If there's an upper bound it won't remove it

Reply:
```
let a : [> `B ] list  = [ ` A ]
```
gives

```
val a : [> `A | `B ] list
```